### ISSUE TITLE
`Style/ClassAndModuleChildren`: `EnforcedStyleForClasses` / `EnforcedStyleForModules` are silently ignored

### ISSUE THREAD

**Issue by bbatsov** [rubocop/rubocop#15321]
The per-type style options (`EnforcedStyleForClasses`, `EnforcedStyleForModules`, added in #12851) don't actually take effect.

With this config:

```yaml
Style/ClassAndModuleChildren:
  EnforcedStyle: compact
  EnforcedStyleForClasses: nested
```

no offense is registered for either class, though both are compact and should be flagged:

```ruby
class Foo::Bar
end

class Foo::Baz < Super
end
# => no offenses detected
```

There are two root causes in `lib/rubocop/cop/style/class_and_module_children.rb`:

1. `style_for_classes` / `style_for_modules` return the raw config **string** (`"nested"`), while every check compares against the **symbol** `:nested`, so `check_style` always routes to the compact branch. (The global `style` from `ConfigurableEnforcedStyle` is `.to_sym`-normalized; these helpers aren't.)
2. The `on_class` guard `return if node.parent_class && style != :nested` reads the global `style` instead of `style_for_classes`, so inherited classes get skipped even once (1) is addressed.

Fixing detection is straightforward, but it exposes a harder problem with autocorrect: nesting `class Foo::Bar` has to invent a namespace wrapper whose kind (class vs module) is unknowable, and when the module style resolves to `compact`, that wrapper is immediately re-compacted - so `rubocop -A` ping-pongs between the two forms. This is the same ambiguity already called out in the cop's `@safety` note. So a complete fix needs a decision on how (or whether) autocorrect should behave for the mixed `compact`/`nested` combination - e.g. register the offense but skip autocorrection when the result can't be determined safely.

Opening this for discussion before sending a PR.
